src/lib/components/Swap/Summary/Details.tsx

```diff
@@ -27,31 +28,42 @@ function Detail({ label, value }: DetailProps) {
 }
 
 interface DetailsProps {
-  input: Currency
-  output: Currency
+  trade: Trade<Currency, Currency, TradeType>
 }
 
-export default function Details({ input, output }: DetailsProps) {
+export default function Details({ trade }: DetailsProps) {
   const integrator = window.location.hostname
 
-  const { maxSlippage } = useAtomValue(settingsAtom)
+  const { allowedSlippage } = useSwapInfo()
+
+  const { inputAmount, outputAmount } = trade
+  const inputCurrency = inputAmount.currency
+  const outputCurrency = outputAmount.currency
+
   const [integratorFee] = useAtom(integratorFeeAtom)
 
+  const priceImpact = useMemo(() => {
+    if (!trade) return undefined
+    const realizedLpFeePercent = computeRealizedLPFeePercent(trade)
+    const priceImpact = trade.priceImpact.subtract(realizedLpFeePercent)
+    return priceImpact
+  }, [trade])
+
   const details = useMemo((): [string, string][] => {
-    // @TODO(ianlapham) = update details to pull derived value from useDerivedSwapInfo
+    // @TODO(ianlapham) check that provider fee is even a valid list item
     return [
       // [t`Liquidity provider fee`, `${swap.lpFee} ${inputSymbol}`],
-      [t`${integrator} fee`, integratorFee && `${integratorFee} ${currencyId(input)}`],
-      // [t`Price impact`, `${swap.priceImpact}%`],
-      // [t`Maximum sent`, swap.maximumSent && `${swap.maximumSent} ${inputSymbol}`],
-      // [t`Minimum received`, swap.minimumReceived && `${swap.minimumReceived} ${outputSymbol}`],
-      [t`Slippage tolerance`, `${maxSlippage}%`],
+      [t`${integrator} fee`, integratorFee && `${integratorFee} ${currencyId(inputCurrency)}`],
+      [t`Price impact`, `${priceImpact?.toFixed(2)}%`],
+      [t`Maximum sent`, `${trade.maximumAmountIn(allowedSlippage).toSignificant(6)} ${inputCurrency.symbol}`],
```

Comment:
L58 and L59 should be mutually exclusive based on what kind of trade you are executing.

Reply:
bump